AppointEase/AppointEase.Application.Contracts/Interfaces/IPatientService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using AppointEase.Application.Contracts.Models.Operations;
+using AppointEase.Application.Contracts.ModelsDto;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AppointEase.Application.Contracts.Interfaces
+{
+    public interface IPatientService
+    {
+        Task <PatientRequest> GetPerson(int patientId);
+        Task <IEnumerable<PatientRequest>> GetAllPersons();
```

Comment:
You are returning an `IEnumerable `of `PatientRequest`? Why are we returning a request object? We need `PatientResponse `objects that we should return...

Reply:
So basically, PersonRequest is mapped to TblPatient, so only the data that needs to be presented will be returned, this is the reason why PersonRequest is returned

Reply:
`PersonRequest` it's a request object, it only maps the values that the front calls us, the naming suggests that. Also our architecture is meant to be calls on 3 different layers, `Rest`-`Application`-`Data`. If we are receiving an object and returning the same object than we do not need layers because we are not separating the concerns of the objects, we are having high-coupling of objects. There should always be 1 object that receives data, we can map it to another object in which there are only data that we need to return.